static/app/components/core/switch/index.tsx

```diff
@@ -2,70 +2,118 @@ import {forwardRef} from 'react';
 import styled from '@emotion/styled';
 
 export interface SwitchProps
-  extends Omit<React.InputHTMLAttributes<HTMLButtonElement>, 'size' | 'type'> {
+  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size' | 'type' | 'onClick'> {
   size?: 'sm' | 'lg';
 }
 
-export const Switch = forwardRef<HTMLButtonElement, SwitchProps>(
+export const Switch = forwardRef<HTMLInputElement, SwitchProps>(
   ({size = 'sm', ...props}: SwitchProps, ref) => {
     return (
-      <SwitchButton
-        ref={ref}
-        data-test-id="switch"
-        type="button"
-        role="checkbox"
-        aria-checked={props.checked}
-        size={size}
-        {...props}
-      >
-        <Toggle disabled={props.disabled} checked={props.checked} size={size} />
-      </SwitchButton>
+      <SwitchWrapper>
+        {/* @TODO(jonasbadalic): if we name the prop size, it conflicts with the native input size prop,
+         * so we need to use a different name, or somehow tell emotion to not create a type intersection.
+         */}
+        <NativeHiddenCheckbox ref={ref} type="checkbox" toggleSize={size} {...props} />
+        <FakeCheckbox size={size}>
+          <FakeCheckboxButton size={size} />
+        </FakeCheckbox>
+      </SwitchWrapper>
```

Comment:
spent a minute looking at [useSwitch from react aria](https://react-spectrum.adobe.com/react-aria/useSwitch.html). It handles some focus things for you. But it also sets a role that seems nice. role="switch" https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/switch/src/useSwitch.ts#L48

Reply:
it seems we’re already somewhat deep into react-aria, so using their switch seems like a good idea, especially since a11y is something we want to keep an eye on.

Reply:
I looked into this, and apparently we dont have that package installed 😅  I'm going to leave a note with this component and leave it up for a future change as I'm fairly sure we'll have more components that could use the functionality from react-aria, and it'd be good to look at all of them